src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseInviteService.cs

```diff
@@ -0,0 +1,230 @@
+                return ValidationActionResult(UserAlreadyHasReleaseRoleInvites);
+            }
+
+            var emailResult = await SendContributorInviteEmail(
+                publicationTitle: publicationTitle,
+                releaseIds: releaseIds,
+                email: email);
+            if (emailResult.IsLeft)
+            {
+                return emailResult;
+            }
+
+            await _userInviteRepository.Create(
+                email: email,
+                role: Role.Analyst,
+                createdById: _userService.GetUserId());
+            await _userReleaseInviteRepository.CreateManyIfNotExists(
+                releaseIds: releaseIds,
+                email: email,
+                releaseRole: Contributor,
+                emailSent: true,
+                createdById: _userService.GetUserId());
+
+            return Unit.Instance;
+        }
+
+        private async Task<Either<ActionResult, Unit>> CreateExistingUserContributorInvite(List<Guid> releaseIds,
+            Guid userId, string email, string publicationTitle)
+        {
+            // check the user doesn't already have the user release roles
+            var existingReleaseRoleReleaseIds = _contentDbContext.UserReleaseRoles
+                .AsQueryable()
+                .Where(urr =>
+                    releaseIds.Contains(urr.ReleaseId)
+                    && urr.Role == Contributor
+                    && urr.UserId == userId)
+                .Select(urr => urr.ReleaseId)
+                .ToList();
+
+            var missingReleaseRoleReleaseIds = releaseIds
+                .Where(releaseId => !existingReleaseRoleReleaseIds.Contains(releaseId))
+                .ToList();
+
+            if (missingReleaseRoleReleaseIds.IsNullOrEmpty())
+            {
+                return ValidationActionResult(UserAlreadyHasReleaseRoles);
+            }
+
+            var emailResult = await SendContributorInviteEmail(
+                publicationTitle: publicationTitle,
+                releaseIds: missingReleaseRoleReleaseIds,
+                email: email);
+            if (emailResult.IsLeft)
+            {
+                return emailResult;
+            }
+
+            await _userReleaseRoleRepository.CreateManyIfNotExists(
+                userId: userId,
+                releaseIds: missingReleaseRoleReleaseIds,
+                role: Contributor,
+                createdById: _userService.GetUserId()
+            );
+
+            await _userReleaseInviteRepository.CreateManyIfNotExists(
+                releaseIds: missingReleaseRoleReleaseIds,
+                email: email,
+                releaseRole: Contributor,
+                emailSent: true,
+                createdById: _userService.GetUserId(),
+                accepted: true);
+
+            return Unit.Instance;
+        }
+
+        private async Task<Either<ActionResult, Unit>> ValidateEmail(string email)
```

Comment:
Don't need this - we can just use the `[EmailAddress]` annotation on the request. This validation is also useless because it doesn't properly check that it's a valid email address.